core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -2050,6 +2053,25 @@ class SocketServerTest {
     }
   }
 
+  @Test
+  def testConnectionDisconnectListenerInvokedOnClose(): Unit = {
+    var listenerConnectionId: String = ""
```

Comment:
Should this be volatile since it's read and written by different threads?

Reply:
Agree, done.